etna/models/tbats.py

```diff
@@ -83,6 +77,154 @@ def get_model(self) -> Model:
         """
         return self._fitted_model
 
+    def forecast_components(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Estimate forecast components.
+
+        Parameters
+        ----------
+        df:
+            features dataframe
+
+        Returns
+        -------
+        :
+            dataframe with forecast components
+        """
+        if self._fitted_model is None or self._freq is None:
+            raise ValueError("Model is not fitted! Fit the model before estimating forecast components!")
+
+        self._check_components()
+
+        horizon = self._get_steps_to_forecast(df=df)
+        raw_components = self._decompose_forecast(horizon=horizon)
+        components = self._select_components(raw_components=raw_components)
+
+        return components
+
+    def predict_components(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Estimate prediction components.
+
+        Parameters
+        ----------
+        df:
+            features dataframe
+
+        Returns
+        -------
+        :
+            dataframe with prediction components
+        """
+        raise NotImplementedError("Prediction decomposition isn't currently implemented!")
+
+    def _get_steps_to_forecast(self, df: pd.DataFrame) -> int:
```

Comment:
Not sure that this is really nessesary to create the separate method for this

Reply:
This is done just to avoid code duplication for sensetive part.

Reply:
Don't get how this method prevent code duplication as you call it only once